pkg/resources/resources.go

```diff
@@ -282,21 +283,30 @@ func KindForObject(scheme *runtime.Scheme, obj runtime.Object) (string, error) {
 	return gvk.Kind, nil
 }
 
-func EnsureGroupVersionKind(s *runtime.Scheme, obj client.Object) error {
-	if obj.GetObjectKind().GroupVersionKind().Kind != "" {
-		return nil
+func GroupVersionKindForObject(s *runtime.Scheme, obj runtime.Object) (schema.GroupVersionKind, error) {
```

Comment:
I'm missing verb in the function name, it sounds as GetGroupVersionKind or ObjectGetGroupVersionKind if namespace should be encoded in the name, as for me. Or it's some common naming convention?

Reply:
No particular reason and I don't think there is a common naming convention, i.e. there is `client.ObjectKeyFromObject` as part of controller-runtime. 

I'll add a `Get` prefix to make it more clear

Reply:
done